test/loader_tests.js

```diff
@@ -59,28 +70,28 @@ describe('component loader', function() {
       },
     };
 
-    let loadA = subject(components)({});
-    let loadB = subject(components)({});
+    let loadA = subject(components);
+    let loadB = subject(components);
 
     let valA = await loadA('test');
     let valB = await loadB('test');
 
-    assume(valA).does.not.equal(valB); 
+    assume(valA).does.not.equal(valB);
   });
   
-  it('should not reinitialize components', async function() {
+  it('should reinitialize components', async () => {
```

Comment:
Last thing to note is this change.

I want it because the `load(component, options)` function from `load = base.loader(componentDirectory, virtualComponents)`, should not hold any state.

This way if we have a stateful component like EC2Manager, or a time based scheduler, expireTask, etc. these components can be reloaded, simply by calling `load(....)`. I think this will be useful in tests, so we don't carry dirty state between tests, but instead a able to easily load fresh components for testing.

If we have use-cases, then in the future we could probably extend `load(component, options)` to take an extra argument `componentsLoaded`, like: `load(component, options, componentsLoaded)`.

Then it could reuse components already loaded in `componentsLoaded` and add components that it loads to `componentsLoaded` (obviously mutating it).. Or something like that...

It could also be `load(component, options, componentsAlreadyLoaded)`, with similar semantics. I can see this as smart, but I don't have a lot of use-cases yet, so maybe let's say we do this later?

I tried to go for minimum we could agree on, then we can extend in the future, I hope we can agree? :)